### PR TITLE
Core/Gossips: Added support for FriendshipFactionID

### PR DESCRIPTION
**Pull request by meji46**
**Changes proposed:**

-  Added support for FriendshipFactionID. It displays a reputation bar at the top of the gossip window

**Issues addressed:**
None

**Tests performed:**
Builds, tested in-game.

**Known issues and TODO list:** (add/remove lines as needed)
None

### CODE REVIEW

**Comment by funjoker** [comment]
why not let GetGossipMenuFriendshipFaction return int32 instead of uint32 ?

File: src/server/game/Globals/ObjectMgr.cpp [hunk 1]
```diff
@@ -9687,6 +9687,48 @@ void ObjectMgr::LoadGossipMenuItems()
...
+    {
+        TC_LOG_INFO("server.loading", ">> Loaded 0 gossip_menu_friendship_faction IDs. DB table `gossip_menu_friendship_faction` is empty!");
+        return;
+    }
+
+    do
+    {
+        Field* fields = result->Fetch();
+
+        uint32 MenuID = fields[0].GetUInt32();
```
Comment (mdX7) [inline src/server/game/Globals/ObjectMgr.cpp:9709]: local variables always lowerCamelCase

Comment (meji46) [reply src/server/game/Globals/ObjectMgr.cpp]: That's what I get for copying the name from the SQL query 😄 Fixed

**Comment by meji46** [comment]
> why not let GetGossipMenuFriendshipFaction return int32 instead of uint32 ?

I was relying on FactionEntry::ID type instead of the packet field. I changed the return of the function to avoid the explicit cast. Thanks!

**Review by mdX7 — APPROVED** [review]

**Comment by Shauren** [comment]
Not bad, I was just having some design thoughts - maybe we shouldn't have named `gossip_menu` (the table holding menu texts) as `gossip_menu` - this way we cant easily have some sort of a "template" table for gossip menu (where this new column could exist) without causing confusion or too much data split

**Pull request merged**